index.bs

```diff
@@ -118,49 +122,64 @@ constantly being pushed from the OS level, at a rate that can be controlled by c
 synchronously, e.g. if it is held by the operating system's in-memory buffers, or asynchronously, e.g. if it has to be
 read from disk. An example pull source is a file handle, where you seek to specific locations and read specific amounts.
 
-Readable streams are designed to wrap both types of sources behind a single, unified interface.
+Readable streams are designed to wrap both types of sources behind a single, unified interface. The implementation
+details of a source are provided by an object containing certain methods and properties that is passed to the
```

Comment:
"containing" may have a specific meaning here. If not I would prefer "having".

The mixture of plural and singular forms in this sentence requires the reader to backtrack to parse it. I'm not sure what the best fix would be, but maybe split it? For example,

> A source is implemented as an object having specific methods and properties. It is passed to the {{ReadableStream()}} constructor.